september/src/main/java/edu/asu/diging/gilesecosystem/september/web/HomeController.java

```diff
@@ -8,27 +8,60 @@
 import org.springframework.ui.Model;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
 
+import edu.asu.diging.gilesecosystem.september.core.datatable.IDataTableData;
+import edu.asu.diging.gilesecosystem.september.core.datatable.impl.DataTableData;
+import edu.asu.diging.gilesecosystem.september.core.db.IMessageDbClient;
 import edu.asu.diging.gilesecosystem.september.core.model.IMessage;
 import edu.asu.diging.gilesecosystem.september.core.service.IMessageManager;
 
 @Controller
 public class HomeController {
-    
+
     @Autowired
     private IMessageManager manager;
 
+    @Autowired
+    private IMessageDbClient dbClient;
+
     @RequestMapping(value = "/")
     public String home(Principal principal, Model model, @RequestParam(defaultValue = "0") int page) {
         if (principal == null || principal.getName() == null || principal.getName().equals("anonymousUser")) {
             return "home";
         }
-        
+
         List<IMessage> messages = manager.getMessages(page);
         model.addAttribute("messages", messages);
         model.addAttribute("totalPages", manager.getNumberOfPages());
         model.addAttribute("currentPageValue", page);
-        
         return "home";
     }
+
+    @RequestMapping(value = "/datatable")
+    public @ResponseBody IDataTableData doGet(@RequestParam int draw, @RequestParam int start, @RequestParam String type)
+            throws Exception {
+        IDataTableData dataTableData = new DataTableData();
+        List<IMessage> dataTableMessages = null;
+        int totalRecords = 0;
+        int filteredRecords = 0;
+        int offset = start / 10;
```

Comment:
this is still hardcoded

Reply:
Addressed it by passing the length parameter from datatable instead of the value 10.

Reply:
if i'm not completely mistaken, this is a configurable parameter and should be taken from the config file.